cranelift/codegen/src/isa/x86/enc_tables.rs

```diff
@@ -596,6 +596,98 @@ fn expand_minmax(
     cfg.recompute_block(pos.func, done);
 }
 
+/// This legalization converts a minimum/maximum operation into a sequence that matches the
+/// non-x86-friendly WebAssembly semantics of NaN handling. This logic is kept separate from
+/// [expand_minmax] above (the scalar version) for code clarity.
+fn expand_minmax_vector(
+    inst: ir::Inst,
+    func: &mut ir::Function,
+    _cfg: &mut ControlFlowGraph,
+    _isa: &dyn TargetIsa,
+) {
+    let ty = func.dfg.ctrl_typevar(inst);
+    debug_assert!(ty.is_vector());
+    let (x, y, x86_opcode, is_max) = match func.dfg[inst] {
```

Comment:
nit: Can we get rid of `is_max` by just looking at the opcode, instead of the one use of `is_max`?

Reply:
We could but then it would be `let (x, y, x86_opcode, clif_opcode)` and we would have to do the opcode comparison twice (once in the `match` and below in the current `if` using `is_max`).

Reply:
I'm going to merge this as-is but I can make a subsequent change to `let (x, y, x86_opcode, clif_opcode)` if you prefer that.